projects/js-packages/publicize-components/src/social-store/actions/connection-data.js

```diff
@@ -202,3 +229,54 @@ export function deleteConnectionById( { connectionId, showSuccessNotice = true }
 		}
 	};
 }
+
+/**
+ * Creates a connection.
+ *
+ * @param {Record<string, any>} data - The data for API call.
+ * @returns {void}
+ */
+export function createConnection( data ) {
+	return async function ( { dispatch } ) {
+		const { createErrorNotice, createSuccessNotice } = coreDispatch( globalNoticesStore );
+
+		try {
+			const path = `/jetpack/v4/social/connections/`;
+
+			dispatch( creatingConnection() );
+
+			const connection = await apiFetch( { method: 'POST', path, data } );
+
+			if ( connection ) {
+				dispatch(
+					addConnection( {
+						...connection,
+						// TODO fix this messy data structure
+						connection_id: connection.ID.toString(),
+						display_name: connection.external_display,
+						service_name: connection.service,
+						external_id: connection.external_ID,
+						profile_link: connection.external_profile_URL,
+						profile_picture: connection.external_profile_picture,
+						can_disconnect: true,
+					} )
+				);
+
+				createSuccessNotice( __( 'Connection successful.', 'jetpack' ), {
```

Comment:
Should we say like "Social account connected successfully" to something to be more explicit?

Reply:
Fixed in https://github.com/Automattic/jetpack/pull/37295/commits/7ede54517923be9f5d33ddbcf53422746ce2bf5e. Thanks